src/Protocol/ProtocolBlockTypePalette.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+#include <unordered_map>
+#include "../BlockState.h"
+
+
+/** Parses and holds a collection of block types and their possible states
+together with their corresponding Id within the Minecraft network protocol. */
+class ProtocolBlockTypePalette
+{
+public:
+	static const UInt32 NOT_FOUND = UINT32_MAX;
+
+	/** Create a new empty instance. */
+	ProtocolBlockTypePalette();
+
+	/** Loads the palette from a string.
+	See loadFromStream() for further details. */
+	bool loadFromString(const AString & aMapping);
+
+	/** Loads the palette from an input stream.
+	Returns `true` on success, `false` otherwise. Sucessive calls to this method
+	will _add_ data to the palette. If duplicate keys are encountered, they will
+	be ignored and an info message logged. */
+	bool loadFromStream(std::istream & aInputStream);
```

Comment:
I keep pushing you to use exceptions, and you keep returning bool :) Any specific reason not to use exceptions?

Reply:
I changed it back, because you objected to the previous implementation ... :confused:

Also the difference between which error case should throw and which should return `false` was unclear.

Anyway unless there is a functional issue, I propose to leave it as is now.

Reply:
I only objected to mixing exceptions with failure-by-return-value. I'd have slightly preferred exceptions all the way, but this is good enough for now, we'll see how it behaves.